clang/lib/Sema/SemaBounds.cpp

```diff
@@ -3998,17 +4027,29 @@ namespace {
       // starts at the beginning of a declaration T v = e, then extra
       // diagnostics may be emitted for T.
       SourceLocation Loc = St->getBeginLoc();
-      if (isa<DeclStmt>(St))
+      SourceRange SrcRange = St->getSourceRange();
+      auto BDCType = Sema::BoundsDeclarationCheck::BDC_Statement;
+      if (isa<DeclStmt>(St)) {
         Loc = V->getLocation();
+        BDCType = Sema::BoundsDeclarationCheck::BDC_Initialization;
+      }
+
+      // Find the last assignment to V to blame in the diagnostic message if it
```

Comment:
The expression in BlameAssignments (if it exists) might not be an assignment to V. It could be an assignment to a variable used in the bounds of V.